lib/ignored-paths.js

```diff
@@ -92,21 +69,81 @@ function IgnoredPaths(patterns) {
  * @returns {IgnoredPaths} object, with patterns loaded from the ignore file
  */
 IgnoredPaths.load = function(options) {
-    var patterns;
+    var ignoredPaths = new IgnoredPaths();
+    ignoredPaths.loadFile(options);
+    return ignoredPaths;
+};
 
-    options = options || {};
+/**
+ * Reset IgnoredPaths
+ * @returns {void}
+ */
+IgnoredPaths.prototype.reset = function() {
+    this.baseDir = null;
+    this.ig = new ignore.Ignore({
+        ignore: this.defaultPatterns
+    });
+
+    this.ig.on("warn", function(e) {
+        var originInfo = (e.data && e.data.origin) ? " (" + e.data.origin + ")" : "";
+        console.info("IgnoredPaths: " + e.message + originInfo);
+    });
+};
 
-    if (options.ignore) {
-        patterns = loadIgnoreFile(options.ignorePath || findIgnoreFile());
-    } else {
-        patterns = [];
-    }
+/**
+ * Load wrapper for IgnorePaths instances
+ * @param {Object} options object containing 'ignore' and 'ignorePath' properties
+ * @returns {void}
+ */
+IgnoredPaths.prototype.loadFile = function(options) {
```

Comment:
Creating a new instance of IgnoredPaths can be achieved without loading a file. Will replace `IgnoredPaths.load(options)` (that returns a new instance) with always using `new IgnoredPath(options)` instead, so that this function can stay private.